source/MRMesh/MRPython.h

```diff
@@ -127,5 +118,61 @@ struct PythonFunctionAdder
     MRMESH_API PythonFunctionAdder( const std::string& moduleName, PyObject* ( *initFncPointer )( void ) );
 };
 
+template<typename R, typename... Args>
+auto decorateExpected( std::function<tl::expected<R, PathError>( Args... )>&& f ) -> std::function<R( Args... )>
+{
+    return[fLocal = std::move( f )]( Args&&... args ) mutable -> R
+    {
+        auto res = fLocal(std::forward<Args>( args )...);
+        if (!res.has_value())
+            switch(res.error())
+            {
+                case MR::PathError::StartEndNotConnected:
+                    throw pybind11::value_error("no path can be found from start to end, because they are not from the same connected component");
+                    break;
+                default:
+                    throw std::runtime_error("please, report the developers for further investigations");
+                    break;
+            }
+        return res.value();
+    };
```

Comment:
mb use template for error type, and have functnions with defined converison to string?

Reply:
done